packages/connection-form/src/utils/set-app-name-if-missing.ts

```diff
@@ -14,7 +20,11 @@ export function setAppNameParamIfMissing(
     const searchParams =
       connectionStringUrl.typedSearchParams<MongoClientOptions>();
     if (!searchParams.has('appName') && defaultAppName !== undefined) {
-      searchParams.set('appName', defaultAppName);
+      const appName = `${defaultAppName}${
+        telemetryAnonymousId ? `-${telemetryAnonymousId}` : ''
+      }-${connectionId}`;
+
+      searchParams.set('appName', appName);
```

Comment:
not related to this line but generally: Should this skip release notes as this pertains to telemetry change?

Reply:
Hm yeah, it's a telemetry change but it's one that could definitely be visible to users, especially non-Atlas users who do on-prem deployments.

Have we considered scoping this change to Atlas only?

Reply:
that's a good idea